lib/valueflow.cpp

```diff
@@ -444,6 +444,18 @@ static void setTokenValue(Token* tok, const ValueFlow::Value &value, const Setti
             return;
         }
 
+        // known result when a operand is true.
+        if (Token::simpleMatch(parent, "&&") && value.isKnown() && value.isIntValue() && value.intvalue==0) {
```

Comment:
hmm.. I see no testcase for these changes in valueflow.cpp. I wonder if you can put these in a separate pull request?

Reply:
I've added test cases for this:

https://github.com/danmar/cppcheck/pull/1294/files#diff-7d93acb765acc90e00196cf9ebe0bec8R562